Client/Frontend/Browser/Tabs/TabTrayV2ViewModel.swift

```diff
@@ -53,17 +53,15 @@ class TabTrayV2ViewModel: NSObject {
     }
 
     func timestampToSection(_ tab: Tab) -> TabSection {
-        let tabDate = Date.fromTimestamp(tab.lastExecutedTime ?? Date.now())
+        let calendar = Calendar.current
         let now = Date()
-
-        if tabDate <= Date().lastWeek {
-            return .older
-        } else if tabDate <= Date.yesterday {
-            return .lastWeek
-        } else if tabDate <= now {
-            return .yesterday
-        } else if tabDate == now {
+        let tabDate = Date.fromTimestamp(tab.lastExecutedTime ?? Date.now())
+        if calendar.isDateInToday(tabDate) {
             return .today
+        } else if calendar.isDateInYesterday(tabDate) {
+            return .yesterday
+        } else if tabDate <= now.lastWeek {
```

Comment:
Ideally we should also do proper comparison for last week using date components just to make it more robust